h2/settings.py

```diff
@@ -253,6 +253,15 @@ def __iter__(self):
     def __len__(self):
         return len(self._settings)
 
+    def __eq__(self, other):
+        if isinstance(other, Settings):
+            return self._settings == other._settings
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        return not self == other
```

Comment:
Ok, this needs to be expanded a little bit. In particular, the `if isinstance` check needs to be repeated here so that we return `NotImplemented` if `other` is not a `Settings` object.